app/models/shipment.rb

```diff
@@ -6,4 +6,9 @@ class Shipment < ApplicationRecord
   validates :origin_country, presence: true
   validates :tracking_number, presence: true, uniqueness: true
   validates :slug, presence: true
+
+  def group_items(count_list=Hash.new(0))
```

Comment:
One possible solution for this

```ruby
shipment_items.select("description").group("description").order('count_description desc').count.map { |k,v| { description: k, count: v} }
```

Reply:
Thank you for the suggestion this seems like a much better way of grouping the items, I will change this. Just to make sure I understand, here you are calling the method .select to select the description then you are using the method .group to group the descriptions that match followed by the method .order which will order the count of descriptions in descending order followed by .map to make it into a new hash.

Reply:
correct

Reply:
Hi, I tried your solution and tested it in my local host and it works fine, however when I run my RSpec test it now fails and returns an empty array instead of the expected result. Are you able to clarify why this happens as I am not sure, also would you like me to try and adjust my test accordingly? I have included the test result below.
 expected: [{:count=>3, :description=>"iPad"}, {:count=>2, :description=>"iPhone"}, {:count=>1, :description=>"Apple Watch"}]
            got: []

Reply:
@MMartins90 feel free to move forward, it's not critical here :)